hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/HAUtils.java

```diff
@@ -389,18 +391,25 @@ public static List<String> buildCAList(CertificateClient certClient,
       if (!SCMHAUtils.isSCMHAEnabled(configuration)) {
         return generateCAList(certClient);
       } else {
+        Set<X509Certificate> rootCAs = certClient.getAllRootCaCerts();
```

Comment:
With the addition of this, I believe the current code will never pass through after the return statement in 400.

We should definitely spend some time clearing out this code with compatibility needs in mind. Can you please file an additional JIRA for this under HDDS-7335 as a sub-task please?